### ISSUE TITLE
Using Locale::CLDR corrupts CLDR::Number

### ISSUE THREAD

**Issue by MarcelVersteeg** [ThePilgrim/perlcldr#36]
In a project I am using `CLDR::Number` for quite some time to format numbers in the right locale. Updating them all to use the number formatters of `Locale::CLDR` would mean a lot of work, I preferrably not want to.

Now I want to use `Locale::CLDR` to get country names in the correct language. However, as soon as I use `Locale::CLDR`, formatting an integer number via `CLDR::Number` fails with the message:
`Can't locate object method "ffround" via package "Math::BigInt" at <path_to>/perllib/CLDR/Number/Role/Format.pm line 260`

I can easily reproduce this using the following script:
```
#!/usr/bin/perl

use strict;

use CLDR::Number;
use Locale::CLDR;

my $cldr = CLDR::Number->new(locale => 'en');
my $formatter = $cldr->decimal_formatter(minimum_fraction_digits => 2, maximum_fraction_digits => 2);
print 'Success: ', $formatter->format(15.23), "\n";
print 'Fail: ', $formatter->format(42.0), "\n";
```

Here, the formatting of the number 42 will fail with the indicated message. As soon as I remove the line `use Locale::CLDR`, the formatting works as expected.

Do you know why using `Locale::CLDR` causes `CLDR::Number` to break? I know that the latter is a somewhat older module, but I do not want to let go of it. If there is a more up-to-date module with a similar interface as `CLDR::Number`, then I will definitely check it out.

**Comment by ehuelsmann** [comment]
The answer is simple: `Locale::CLDR::Locales::*` modules `use bignum;`. And the `bignum` module overwrites `$Math::BigInt::upgrade` as well as `$Math::BigFloat::downgrade`.

While doing so, it doesn't just break `CLDR::Number`; it breaks every project that tries to install its own number handling variant *as well as* every project which tries to disable up- and downgrading.

**Comment by ehuelsmann** [comment]
@MarcelVersteeg please check the 0.34.2 release. It should solve this issue.

**Comment by ehuelsmann** [comment]
@ThePilgrim this is the transcript I have now:

```
$ ./test.pl
Versions:
  CLDR::Number: 0.19
  Locale::CLDR: v0.34.2

Success: 15.23
Fail: 42.00
$
```

The test script is a variant of @MarcelVersteeg's:

```perl
#!/usr/bin/perl

use strict;

use CLDR::Number;
use Locale::CLDR;

print "Versions:\n"; 
print "  CLDR::Number: $CLDR::Number::VERSION\n";
print "  Locale::CLDR: $Locale::CLDR::VERSION\n";
print "\n";

my $cldr = CLDR::Number->new(locale => 'en');
my $formatter = $cldr->decimal_formatter(minimum_fraction_digits => 2, maximum_fraction_digits => 2);
print 'Success: ', $formatter->format(15.23), "\n";
print 'Fail: ', $formatter->format(42.0), "\n";
```



Conclusion: This issue is fixed in 0.34.2.

**Comment by ehuelsmann** [comment]
Thanks for the confirmation!
